bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -175,7 +127,10 @@ public ObjectId() {
      * @param date the date
      */
     public ObjectId(final Date date) {
-        this(dateToTimestampSeconds(date), MACHINE_IDENTIFIER, PROCESS_IDENTIFIER, NEXT_COUNTER.getAndIncrement(), false);
+        this.timestamp = dateToTimestampSeconds(date);
```

Comment:
Can this just be

    this(dateToTimestampSeconds(date), NEXT_COUNTER.getAndIncrement() & LOW_ORDER_THREE_BYTES, false);

Reply:
Yes, and I have made the change.